azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -202,7 +217,29 @@ def should_report(is_healthy, error_state, last_timestamp, period):
         return datetime.datetime.utcnow() >= (last_timestamp + period)
 
     def put_vm_log(self, content):
-        raise NotImplementedError("Unimplemented")
+        """
+        Try to upload VM logs, a compressed zip file, via the host plugin /vmAgentLog channel.
+        :param content: the binary content of the zip file to upload
+        """
+        if not self.ensure_initialized():
+            raise ProtocolError("HostGAPlugin: HostGAPlugin is not available")
+
+        if content is None or self.container_id is None or self.deployment_id is None:
+            raise ProtocolError("HostGAPlugin: Invalid arguments passed to upload VM logs. "
+                                "Content: {0}, container id: {1}, deployment id: {2}".format(content,
+                                                                                             self.container_id,
+                                                                                             self.deployment_id))
+
+        url = URI_FORMAT_PUT_LOG.format(self.endpoint, HOST_PLUGIN_PORT)
+        response = restutil.http_put(url,
```

Comment:
Can http_put raise?

Reply:
It can, thanks for catching that. I added a catch for that (and raise `ProtocolError`) in the caller of this method.

Reply:
the caller coming on a different PR?